### ISSUE TITLE
max_similarity fails when results are excluded due to "pos" argument

### ISSUE THREAD

**Issue by cOnjuer** [alvations/pywsd#27]
```
max_similarity(context_sentence="art entertainment hobby creative art", 
                         ambiguous_word='creative', option="path", lemma=True, 
                         context_is_lemmatized=True, pos='n', best=True)
```

The above function call throws an exception

```
IndexError                                Traceback (most recent call last)
<ipython-input-63-3333bb3d5eca> in <module>()
----> 1 max_similarity(context_sentence="art entertainment hobby creative art", ambiguous_word='creative', option="path", lemma=True, context_is_lemmatized=True, pos='n', best=True)

/root/anaconda2/lib/python2.7/site-packages/pywsd/similarity.pyc in max_similarity(context_sentence, ambiguous_word, option, lemma, context_is_lemmatized, pos, best)
    106         result = sorted([(v,k) for k,v in result.items()],reverse=True)
    107     ##print result
--> 108     if best: return result[0][1];
    109     return result
    110 

IndexError: list index out of range
```

It works when i remove the pos argument.

**Comment by alvations** [comment]
This is because the word "creative" doesn't have a noun synset in wordnet:

```python
>>> from nltk.corpus import wordnet as wn
>>> wn.synsets('creative')
[Synset('creative.a.01'), Synset('creative.s.02')]
```

But it's true that the error message should be more informative, rather than throwing an `IndexError`

**Comment by melOncholic** [comment]
Yes I was expecting an empty result but unfortunately I ran into unhandled exception. Let me know if I can contribute

**Comment by alvations** [comment]
This should be fixed now in the new version `pywsd>=1.2.0`. It should be better with the recent overhaul and the None-y outputs will fallback to zero. Also with more control of the new wordnet reader instead of the NLTK. 

Please do re-open the issue if it comes up =)

Thanks @cOnjuer for raising the issue!
